packages/fxa-settings/src/pages/ResetPassword/container.tsx

```diff
@@ -0,0 +1,177 @@
+              // This becomes a query param (service=sync) on the email link.
+              // We need to modify this in FXA-7657 to send the `client_id` param
+              // when we work on the OAuth flow.
+              ...(service &&
+                service === MozServices.FirefoxSync && { service: 'sync' }),
+            },
+          },
+        });
+        setBeginResetPasswordError(undefined);
+      } catch (error) {
+        // ❌ All error processing should occur here. Let the presentation component
+        // handle just the presentation.
+        const graphQLError: GraphQLError = error.graphQLErrors[0];
+        if (graphQLError && graphQLError.extensions) {
+          const { errno, retryAfterLocalized } = graphQLError.extensions;
+
+          setBeginResetPasswordError({
+            errno,
+            message: AuthUiErrorNos[errno].message,
+            // ❌ Because grabbing the FTL ID from an error always requires a call to
+            // composeAuthUiErrorTranslationId, it seemed best to do that here when
+            // creating a new error object. Allow actual l10n with FtlMsg or FtlMsgResolver
+            // to occur in the presentation layer.
+            ftlId: composeAuthUiErrorTranslationId({ errno }),
+            retryAfterLocalized,
+          });
+        } else {
+          // ❌ Prior to our Account refactor, we had a custom `useMutation` hook that
+          // would capture networkErrors in Sentry. We may want to use that here as well.
+          // https://github.com/mozilla/fxa/blob/690a6b31f787e1325f13776f3ab577f153f1f8b1/packages/fxa-settings/src/lib/hooks.tsx#L68
+          // See our docs that explain more:
+          // https://github.com/mozilla/fxa/tree/main/packages/fxa-settings#gql-error-handling
+
+          // TODO: why is `errno` in `AuthServerError` possibly undefined?
+          // might want to grab from `ERRORS.UNEXPECTED_ERROR` instead
+          const { errno = 999, message } = AuthUiErrors.UNEXPECTED_ERROR;
+          setBeginResetPasswordError({
+            errno,
+            message,
+            ftlId: composeAuthUiErrorTranslationId({ errno }),
+          });
+        }
+      }
+    },
+    [beginResetPassword]
+  );
+
+  // ❌ We may or may not want this per container component but we can show a loading
+  // spinner here if validation is required and hasn't occured yet.
+  if (queryParams === null) {
+    return <LoadingSpinner />;
+  }
+  // ❌ Handle query param validation errors, e.g. return LinkDamaged or whatever
+  // makes sense for that error. Can also access errors like `queryParamErrors.email`
+  if (queryParamErrors) {
+    console.log('queryParamErrors', queryParamErrors);
+  }
+
+  // ❌ Create a result object per query or mutation whose name matches
+  // the handler. On more complex pages we will be passing multiple handlers
+  // and their results.
+  const beginResetPasswordResult: BeginResetPasswordResult = {
```

Comment:
Is there a reason the beginResetPasswordHandler doesn't just return a result?

Reply:
@dschom Good question - when I created the first iteration of this approach, I'm pretty sure it was necessary and/or the separation felt right. I'll revisit this.

Reply:
Circled back here and on my first iteration, I was doing some error handling outside of the `catch` (IIRC I thought there was more complexity than there was) and I needed `setBeginResetPasswordError` to keep track of the error state so I didn't immediately return it from the handler. I didn't check on this again since we similarly need to wait on `beginResetPasswordMutationResult` to update after `await beginResetPassword` is called but I think I can simplify this and then not have to `useEffect` in the page component to reference the data/error which felt awkward anyway, so thanks for the call out. Working on this now, but not sure I will spend much time updating tests to match unless I have time at the end of the sprint since we can do that when we implement this during the reset PW redesign.

Reply:
@dschom This is a bit tricky since `ResetPassword` needs the original state, an update on callback call (for loading state), then an update on callback complete. Something like this...

```
  const [doSomething,  doSomethingResult] = useMutation(DO_SOMETHING);

  const someHandler = useCallback(async(email) => {
    await doSomething({
      variables: {
        email,
      },
    });

    return doSomethingResult;
  });
```

... of course just returns the original `doSomethingResult` state. I can `const { data } = await doSomething` to retrieve `data`, but I'd need to manually `return { data, loading: false, error: undefined }` and do something similar when returning in every `catch` situation which isn't as nice as being able to reference `doSomethingResult` values, and `loading` wouldn't get set to `true` while the call is occuring. I tried manually managing the loading state but that would need to be passed into `ResetPassword` and at that point, I feel we might as well just pass in the entire result object. 

I don't mind the `useEffect` in ResetPassword as much either after playing around with this. If I had more time I would keep digging since maybe it's doable - maybe we can revisit this during implementation?

Reply:
I'm not sure if this is exactly what you had in mind but it made me realize we can just do something like `setResetPasswordLoading(true);` before the call and `setResetPasswordLoading(false);` after. We can also just return `error` or `data` if we're OK with the other being `undefined`. @dschom @chenba I went ahead and pushed that update that's testable locally but didn't adjust tests.

Part of me still likes the idea of using `beginResetPasswordResult` from the mutation for consistency across our components and taking some of the manual state management work out, but, we already need to modify the shape of `error` that's returned in the container component instead of using `beginResetPasswordResult.error`, so this is probably better, and then we don't need the `useEffect` in ResetPassword to get those state updates.

Reply:
> I'm not sure if this is exactly what you had in mind ...

Yep something like that.

I do wish we have actual types to work with but at least destructuring into `{ data, error }` is familiar and easy to understand.

Reply:
Not sure if you mean types from apollo-client but these do have types:

<img src="https://github.com/mozilla/fxa/assets/13018240/a93081bf-47c6-4b7d-b5c7-cdaea37ee5a5" width="420">
<img src="https://github.com/mozilla/fxa/assets/13018240/2bc7a6ce-33fe-4459-b0b2-3568dd5380c1" width="420">

Reply:
heh that was just me doing old-man-yelling-at-cloud.  I meant types at runtime and able to pattern match on them.  Tagged union types are possible in Typescript, but that's still matching on the tag strings.